src/main.ts

```diff
@@ -302,110 +324,164 @@ async function getTsServerRepoResult(
             }
         }
 
-        const owner = repo.owner ? `${mdEscape(repo.owner)}/` : "";
-        const url = repo.url ? `(${repo.url})` : "";
+        const tsServerResult = {
+            oldServerFailed,
+            oldSpawnResult,
+            newServerFailed,
+            newSpawnResult,
+            replayScriptPath,
+            installCommands,
+        };
 
-        let summary = `## [${owner}${mdEscape(repo.name)}]${url}\n`;
+        if (oldServerFailed && !newServerFailed) {
+            return { status: "Detected interesting changes", tsServerResult }
+        }
+        if (!newServerFailed) {
+            return { status: "Detected no interesting changes" };
+        }
 
-        if (oldServerFailed) {
-            const oldServerError = oldSpawnResult?.stdout
-                ? prettyPrintServerHarnessOutput(oldSpawnResult.stdout, /*filter*/ true)
-                : `Timed out after ${executionTimeout} ms`;
-            summary += `
+        return { status: "Detected interesting changes", tsServerResult, replayScriptPath, rawErrorPath };
+    }
+    catch (err) {
+        reportError(err, `Error running tsserver on ${repo.url ?? repo.name}`);
+        return { status: "Unknown failure" };
+    }
+    finally {
+        console.log(`Done ${repo.url ?? repo.name}`);
+        logStepTime(diagnosticOutput, repo, "language service", lsStart);
+    }
+}
+
+function groupOldErrors(summaries: Summary[]): Map<string, Summary[]> {
+    const group = new Map<string, Summary[]>();
+    for (const summary of summaries) {
+        if (summary.tsServerResult.oldServerFailed && !summary.tsServerResult.newServerFailed) {
+            const error = parseServerHarnessOutput(summary.tsServerResult.oldSpawnResult!.stdout);
+
+            const key = typeof error === "string" ? getHash([error]) : getHashForStack(error.message);
+            const value = group.get(key) ?? [];
+            value.push(summary);
+            group.set(key, value);
+        }
+    }
+
+    return group;
+}
+
+function groupNewErrors(summaries: Summary[]): Map<string, Summary[]> {
+    const group = new Map<string, Summary[]>();
+    for (const summary of summaries) {
+        if (summary.tsServerResult.newServerFailed) {
+            const error = parseServerHarnessOutput(summary.tsServerResult.newSpawnResult!.stdout);
+
+            const key = typeof error === "string" ? getHash([error]) : getHashForStack(error.message);
```

Comment:
minor: it's odd to have `getErrorMessage` which is only called once but not for this pattern (or vice versa, maybe they should all be inline)

Reply:
This is actually a bug. `getErrorMessage` should be called for the old server errors as well. Fixed.